homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -361,6 +403,23 @@ async def async_setup_entry(
             EnvoyConsumptionEntity(coordinator, description)
             for description in CONSUMPTION_SENSORS
         )
+    # For each production phase reported add production entities
+    if envoy_data.system_production_phases:
+        entities.extend(
+            EnvoyProductionPhaseEntity(coordinator, description)
+            for use_phase, phase in envoy_data.system_production_phases.items()
+            for description in PRODUCTION_PHASE_SENSORS[PhaseNames(use_phase)]
+            if phase is not None
+        )
+    # For each consumption phase reported add consumption entities
+    if envoy_data.system_consumption_phases:
+        entities.extend(
+            EnvoyConsumptionPhaseEntity(coordinator, description)
+            for use_phase, phase in envoy_data.system_consumption_phases.items()
+            for description in CONSUMPTION_PHASE_SENSORS[PhaseNames(use_phase)]
```

Comment:
I think we could simplify this further and make `use_phase` the key to it becomes `CONSUMPTION_PHASE_SENSORS[use_phase]` and the dict construction above is adjusted accordingly

Sorry for missing this one the first go.

Reply:
The linter didn't like the  
```py
for description in CONSUMPTION_PHASE_SENSORS[use_phase]
```
when the walrus was added and the `PhaseNames` had to be added. To optimize it we need to change the definition in the EnvoyConsumptionRequiredKeysMixin to str

```py
    on_phase: str | None
```
and make sure the key is str
```py
CONSUMPTION_PHASE_SENSORS = {
    (on_phase := str(PhaseNames(PHASENAMES[phase]))): [
```

Reply:
No worries. Let's leave it as is